src/domains/sign/api_product/sign/_base_policy.xml

```diff
@@ -3,23 +3,76 @@
-                <set-header
-                    name="X-Forwarded-For"
-                    exists-action="override"
-                >
+                <cache-lookup-value key="@(context.Subscription.Id+"_whitelist")" variable-name="issuerWhitelist" />
+                <choose>
+                    <!-- Check if the IP whitelist is in cache -->
+                    <when condition="@(!context.Variables.ContainsKey("issuerWhitelist"))">
+                        <!-- Executes a query on CosmosDb via REST API to obtain the list of whitelisted IPs for single issuer (subscriptionId) -->
+                        <set-variable name="requestDateString" value="@(DateTime.UtcNow.ToString("r"))" />
+                        <send-request mode="new" response-variable-name="response" timeout="10" ignore-error="false">
+                            <set-url>https://{{io-sign-cosmosdb-name}}.documents.azure.com/dbs/{{io-sign-cosmosdb-issuer-container-name}}/colls/{{io-sign-cosmosdb-issuer-whitelist-ip-collection-name}}/docs</set-url>
+                            <set-method>POST</set-method>
+                            <set-header name="Authorization" exists-action="override">
+                                <value>@{
+                                    var verb = "post";
+                                    var resourceType = "docs";
+                                    var resourceLink = "dbs/{{io-sign-cosmosdb-issuer-container-name}}/colls/{{io-sign-cosmosdb-issuer-whitelist-ip-collection-name}}";
+                                    var key = "{{io-sign-cosmosdb-key}}";
+                                    var keyType = "master";
+                                    var tokenVersion = "1.0";
+                                    var date = context.Variables.GetValueOrDefault<string>("requestDateString");
+
+                                    var hmacSha256 = new System.Security.Cryptography.HMACSHA256 { Key = Convert.FromBase64String(key) };
+
+                                    string payLoad = string.Format("{0}\n{1}\n{2}\n{3}\n{4}\n",
+                                            verb.ToLowerInvariant(),
+                                            resourceType.ToLowerInvariant(),
+                                            resourceLink,
+                                            date.ToLowerInvariant(),
+                                            ""
+                                    );
+
+                                    byte[] hashPayLoad = hmacSha256.ComputeHash(System.Text.Encoding.UTF8.GetBytes(payLoad));
+                                    string signature = Convert.ToBase64String(hashPayLoad);
+
+                                    return System.Uri.EscapeDataString(String.Format("type={0}&ver={1}&sig={2}",
+                                        keyType,
+                                        tokenVersion,
+                                        signature));
+                                    }</value>
```

Comment:
can you comment this part or link a reference to understand what's happening here and why we're computing an hash?